### PR TITLE
Better search user signature

### PR DESCRIPTION
**Pull request by aviadl**
fixes https://github.com/descope/etc/issues/5137
+test

### CODE REVIEW

File: lib/management/user.ts [hunk 1]
```diff
@@ -324,6 +337,30 @@ const withUser = (sdk: CoreSdk, managementKey?: string) => ({
...
     ),
+  /**
+   * Search all users. Results can be filtered according to tenants and/or
+   * roles, and also paginated used the limit and page parameters.
+   * @param tenantIds optional list of tenant IDs to filter by
+   * @param roles optional list of roles to filter by
+   * @param limit optionally limit the response, leave out for default limit
+   * @param page optionally paginate over the response
+   * @param testUsersOnly optionally filter only test users
+   * @param withTestUser optionally include test users in search
```
Comment (shilgapira) [inline lib/management/user.ts:348]:
```suggestion
   * @param searchReq an object with all the constraints for this search
```

**Review by shilgapira — DISMISSED** [review]

**Comment by asafshen** [comment]
@aviadl should we add to readme?

also I think we should deprecate `searchAll` (mention it as deprecated) so:
a) developers using the sdk won't be confused about what should they use
b) sdk developers won't copy this pattern (this will not guaranteed that the pattern won't copy, but might help a bit imo)

File: lib/management/user.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { SdkResponse, transformResponse, UserResponse, LoginOptions } from '@descope/core-js-sdk';
+import { deprecate } from 'util';
```
Comment (asafshen) [inline lib/management/user.ts:2]: great 👍

**Review by asafshen — APPROVED** [review]
Great!

**Pull request merged**